containerbuddy/config.go

```diff
@@ -65,6 +66,7 @@ type ServiceConfig struct {
 	TTL              int             `json:"ttl"`
 	Interfaces       json.RawMessage `json:"interfaces"`
 	Tags             []string        `json:"tags,omitempty"`
+	Endpoint         string          `json:"endpoint,omitempty"`
```

Comment:
The `endpoint` name seems ambiguous or possibly misleading. In this context, it's the hostname that you want Containerbuddy to resolve to an IP before registering the service in Consul.